kinto/core/authorization.py

```diff
@@ -171,7 +171,10 @@ def check_permission(self, principals, bound_perms):
         if not bound_perms:
             bound_perms = [(self.resource_name, self.required_permission)]
         for (_, permission) in bound_perms:
-            setting = '{}_{}_principals'.format(self.resource_name, permission)
+            if self.resource_name and permission.startswith(self.resource_name):
+                setting = '{}_principals'.format(permission.replace(':', '_'))
+            else:
+                setting = '{}_{}_principals'.format(self.resource_name, permission)
```

Comment:
Does it apply to `collection_create_principal`  to? :open_mouth:

Reply:
Yes probably, we should investigate that too.